### PR TITLE
Add workaround for VS2013 const-incorrect atomic

### PR DESCRIPTION
**Pull request by BillyONeal**

### CODE REVIEW

File: src/benchmark.cc [hunk 1]
```diff
@@ -115,7 +115,13 @@ std::string* GetReportLabel() {
 
 // Global variable so that a benchmark can report an error as a human readable
 // string. If error_message is null no error occurred.
-static std::atomic<const char*> error_message = ATOMIC_VAR_INIT(nullptr);
+#if defined(BENCHMARK_WORKAROUND_VS_2013)
```
Comment (dominichamon) [inline src/benchmark.cc]: in this case, it's probably worth just keeping the `#if _MSC_VER <= 1800` here (with a comment explaining it). the extra macro doesn't add anything useful.

Comment (BillyONeal) [inline src/benchmark.cc]: I was concerned about exposing other compilers to the _MSC_VER check; will
```
#if defined(_MSC_VER) && _MSC_VER <= 1800
```
work?

Comment (dominichamon) [inline src/benchmark.cc]: it should do.

`#if _MSC_VER <= 1800` alone won't as `_MSC_VER` will be 0 if undefined.

**Comment by BillyONeal** [comment]
OK, we'll see if Travis says the other compilers are happy

**Comment by BillyONeal** [comment]
https://ci.appveyor.com/project/BillyONeal/benchmark/build/45 AppVeyor shows MSVC-12 is happy so far

**Comment by dominichamon** [comment]
https://ci.appveyor.com/project/google/benchmark/build/205 agrees!

(i can't make it comment on the PR for some reason)

**Comment by BillyONeal** [comment]
Guess I set that up in my own fork for giggles then :)

**Comment by dominichamon** [comment]
thanks!

**Pull request merged**